src/custom/pages/Profile/index.tsx

```diff
@@ -63,8 +73,46 @@ export default function Profile() {
   const vCowBalanceUnvested = formatSmartLocaleAware(unvested, AMOUNT_PRECISION) || '0'
   const vCowBalance = formatSmartLocaleAware(total, AMOUNT_PRECISION) || '0'
 
-  const hasCowBalance = !cow?.equalTo(0)
-  const hasVCowBalance = !total?.equalTo(0)
+  const hasCowBalance = cow && !cow?.equalTo(0)
+  const hasVCowBalance = total && !total?.equalTo(0)
+  const hasVestedBalance = vested && !vested?.equalTo(0)
+
+  // Init modal hooks
+  const { handleSetError, handleCloseError, ErrorModal } = useErrorModal()
+  const { TransactionConfirmationModal, openModal, closeModal } = useTransactionConfirmationModal(
+    OperationType.ORDER_SIGN
+  )
+
+  // Boolean flags
+  const isSwapPending = swapVCowStatus === SwapVCowStatus.SUBMITTED
+  const isSwapInitial = swapVCowStatus === SwapVCowStatus.INITIAL
+  const isSwapDisabled = Boolean(!hasVestedBalance || !isSwapInitial || isSwapPending)
+
+  // Handle swaping
+  const { swapCallback } = useSwapVCowCallback({
+    openModal,
+    closeModal,
+  })
+
+  const handleVCowSwap = useCallback(async () => {
+    handleCloseError()
+
+    if (!swapCallback) {
+      return
+    }
+
+    setSwapVCowStatus(SwapVCowStatus.ATTEMPTING)
```

Comment:
I already mentioned in some other PR but i was not 100% sure about this approach. 

Why a global unique status? Are we sure we want to take this approach? In principle you can send a claimAll and before is mined send another one.

I guess you are right this approach is simpler, only allowing the user to do one at a time, but we could just do the same thing and have a hook that return if there's any claimAll going on